libs/cloud-database/src/database.rs

```diff
@@ -70,11 +74,34 @@ impl CloudDatabase {
     #[instrument(skip(self))]
     pub async fn user_login(&self, login: UserLogin) -> Result<Option<UsersModel>, DbErr> {
         info!("database user_login enter");
-        Users::find()
+        let Ok(user) = Users::find()
+            .column(UsersColumn::Password)
             .filter(UsersColumn::Email.eq(login.email))
-            .filter(UsersColumn::Password.eq(login.password))
             .one(&self.pool)
-            .await
+            .await else {
+                return Err(DbErr::RecordNotFound(
+                    "Failed to find the user".to_string(),
+                ));
+            };
+
+        let hashed_password = user
+            .clone()
+            .ok_or(DbErr::RecordNotFound("Failed to find the user".to_string()))?
+            .password
+            .ok_or(DbErr::RecordNotFound("Failed to find the user".to_string()))?;
+
+        let parsed_hash = PasswordHash::new(&hashed_password).or(Err(DbErr::RecordNotFound(
```

Comment:
need a fallback for old password or add a migration to renew password already in database: https://github.com/toeverything/OctoBase/tree/master/libs/cloud-database/migration/src

Reply:
I'm sorry but since I do not have access to the firebase, could you please help me to verify that the migration is correct?
After your verfication, I will convert it from draft to normal pull request.